File: testers/avro/java-quarkus/generated-and-checked-in/com/example/events/InvoicePublisher.java
package com.example.events;

import io.smallrye.mutiny.Uni;
import io.smallrye.reactive.messaging.MutinyEmitter;
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.inject.Inject;
import org.eclipse.microprofile.reactive.messaging.Channel;

/** Type-safe event publisher for invoice topic */
@ApplicationScoped
public record InvoicePublisher(
    @Channel("invoice") MutinyEmitter<Invoice> kafkaTemplate, String topic) {
  @Inject
  public InvoicePublisher {}

  public InvoicePublisher(@Channel("invoice") MutinyEmitter<Invoice> kafkaTemplate) {
    this(kafkaTemplate, "invoice");
  }

  public InvoicePublisher withKafkaTemplate(MutinyEmitter<Invoice> kafkaTemplate) {
    return new InvoicePublisher(kafkaTemplate, topic);
  }

  public InvoicePublisher withTopic(String topic) {
    return new InvoicePublisher(kafkaTemplate, topic);
  }

  /** Publish a Invoice event */
  public Uni<Void> publish(String key, Invoice event) {
    return kafkaTemplate.send(event);
  }
}

File: testers/avro/kotlin-quarkus-mutiny/generated-and-checked-in/com/example/events/InvoicePublisher.kt
package com.example.events

import io.smallrye.mutiny.Uni
import io.smallrye.reactive.messaging.MutinyEmitter
import jakarta.enterprise.context.ApplicationScoped
import jakarta.inject.Inject
import java.lang.Void
import org.eclipse.microprofile.reactive.messaging.Channel

@ApplicationScoped
/** Type-safe event publisher for invoice topic */
data class InvoicePublisher @Inject constructor(
  @field:Channel("invoice") val kafkaTemplate: MutinyEmitter<Invoice>,
  val topic: kotlin.String = "invoice"
) {
  /** Publish a Invoice event */
  fun publish(
    key: kotlin.String,
    event: Invoice
  ): Uni<Void> {
    return kafkaTemplate.send(event)
  }
}